### ISSUE TITLE
ZFS glitching on backup server after one-by-one drive swap on raidz1 pool

### ISSUE THREAD

**Issue by darkpixel** [openzfs/zfs#9768]
First off, I'm sorry for the terrible bug report title.
I have no clue what's going on or how to describe it.

### System information
Distribution Name	| Proxmox
Distribution Version	| 6.0-4/2a719255 (running kernel: 5.0.15-1-pve)
Linux Kernel	| 5.0.15-1-pve
Architecture	| x86_64
ZFS Version	| zfs-0.8.1-pve1
SPL Version	| 0.8.1-pve1

### Describe the problem you're observing
zpool status is reporting all sorts of problems after replacing disks.

### Describe how to reproduce the problem
I had a backup server with 7x Western Digital 1TB drives.
As backup usage started increasing, I needed more space.
One by one I ran ```zpool replace -o ashift=9 tank <old-drive> /dev/disk/by-id/ata-WDC_WD40....```
I used ```ashift=9``` because running 'replace' without it would result in an error:
```cannot replace ata-WDC_WD1003FBYZ-010FB0_WD-WCAW37616709 with /dev/disk/by-id/ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCUV1: new device has a different optimal sector size; use the option '-o ashift=N' to override the optimal size```

After all the drives were replaced, my pool size did not grow.
I ran ```zpool set autoexpand=on tank```, and it did not grow.
I had two SSH sessions open, one where I was issuing commands, the other was running ```watch zpool status -v```.  (The watch command runs the supplied arguments every 2 seconds and displays their output.)
I read the man page and figured I should try running ```zpool online -e tank /dev/disk/by-id/ata-WDC.....```

I did that one by one for each drive in the pool.  After I ran the last drive, I flipped over to the ```zpool status -v``` ssh session.

Every time it would run, I would get a different output.

```
  pool: tank
 state: DEGRADED
status: One or more devices is currently being resilvered.  The pool will
        continue to function, possibly in a degraded state.
action: Wait for the resilver to complete.
  scan: resilver in progress since Mon Dec 23 21:06:45 2019
        49.3M scanned at 49.3M/s, 14.9M issued at 14.9M/s, 294G total
        0B resilvered, 0.00% done, 0 days 05:36:05 to go
config:

        NAME                                          STATE     READ WRITE CKSUM
        tank                                          DEGRADED     0     0     0
          raidz1-0                                    DEGRADED     0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1SFLASR  ONLINE       0     0 4.28M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5VCJDZ3  ONLINE       0     0 4.26M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5NKNA80  ONLINE       0     0 4.25M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7JY6ESV  ONLINE       0     0 4.26M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCXHP  ONLINE       0     0 4.30M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1PTT198  ONLINE       0     0 4.32M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCUV1  UNAVAIL      0     0     0

errors: Permanent errors have been detected in the following files:

        tank/backups/usrbgof/officeshare@zrep-remote_000007:<0x0>
        <snip a lot of files listed with errors>
```
Notice that it says the last drive is unavailable.  It's in the system, no errors in dmesg, and smartctl shows it's healthy.  Running ```zpool online -e tank ....``` again on the drive says:

```
warning: device '/dev/disk/by-id/ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCUV1' onlined, but remains in faulted state
use 'zpool replace' to replace devices that are no longer present
```

Anyways, when ```zpool status -v``` is run again, it outputs the following:
```
  pool: tank
 state: DEGRADED
status: One or more devices has experienced an error resulting in data
        corruption.  Applications may be affected.
action: Restore the file in question if possible.  Otherwise restore the
        entire pool from backup.
   see: http://zfsonlinux.org/msg/ZFS-8000-8A
  scan: resilvered 51.5K in 0 days 00:00:05 with 2125 errors on Mon Dec 23 21:06:50 2019
config:

        NAME                                          STATE     READ WRITE CKSUM
        tank                                          DEGRADED     0     0     0
          raidz1-0                                    DEGRADED     0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1SFLASR  ONLINE       0     0 4.28M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5VCJDZ3  ONLINE       0     0 4.27M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5NKNA80  ONLINE       0     0 4.26M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7JY6ESV  ONLINE       0     0 4.26M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCXHP  ONLINE       0     0 4.30M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1PTT198  ONLINE       0     0 4.32M
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCUV1  UNAVAIL      0     0     0

errors: Permanent errors have been detected in the following files:
<snip long list of files>
```

Most of the files listed as being corrupted appear to be from active or recent zfs receives.
The CKSUM column in zpool status is slowly climbing.

I think the reason the ```zpool status``` outputs differ every few seconds is that the resilver seems to be scanning for a few seconds and then stopping and restarting.

I haven't restarted the box yet as it's not exactly mission-critical and I figured someone might want me to gather more information.  I do have a deadline in 24-hours that requires me to restart it and resume backups for year-end.  If the pool is actually hosed, I will destroy it and try to transfer ~20 TB over the internet between now and January 2nd...  No fun. ;)

**Comment by darkpixel** [comment]
The strangeness kept occurring right up until the reboot.
The reboot appears to have fixed it.
The CKSUM errors changed after the reboot, and the errors listed below are not growing.
```
  pool: tank
 state: ONLINE
status: One or more devices has experienced an unrecoverable error.  An
        attempt was made to correct the error.  Applications are unaffected.
action: Determine if the device needs to be replaced, and clear the errors
        using 'zpool clear' or replace the device with 'zpool replace'.
   see: http://zfsonlinux.org/msg/ZFS-8000-9P
  scan: resilvered 49.9M in 0 days 00:00:50 with 0 errors on Tue Dec 24 13:45:08 2019
config:

        NAME                                          STATE     READ WRITE CKSUM
        tank                                          ONLINE       0     0     0
          raidz1-0                                    ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1SFLASR  ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5VCJDZ3  ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K5NKNA80  ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7JY6ESV  ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCXHP  ONLINE       0     0     0
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K1PTT198  ONLINE       0     0   293
            ata-WDC_WD40EFRX-68N32N0_WD-WCC7K7YNCUV1  ONLINE       0     0    49  

errors: No known data errors
```
Expected amount of free space:
```
root@uswuxsdbkp02:~# zfs list tank
NAME   USED  AVAIL     REFER  MOUNTPOINT
tank   106G  20.9T     37.5K  /tank
root@uswuxsdbkp02:~# 
```

**Comment by Ornias1993** [comment]
@darkpixel Some interesting things to add:
- Did you test the disks thoroughly before one-by-one driveswap?
- Are they from the same batch?

**Comment by darkpixel** [comment]
I ran the WD drive diagnosts short test on them, so the answer is 'not really'.
They were ordered at the same time, so there's a chance they are from the same batch.  The serial numbers are wildly different though.

The drives don't appear to be bad at all.  After a reboot, it finished resilvering and displayed no errors.  I ran a scrub and it returned no errors.  I've been dumping data to them like crazy over the last ~24 hours and there are no errors listed by ```zpool status``` and no errors in dmesg.  Smart tests show no problems...so I don't think this is a drive issue.
